Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/AlertMessage.java

```diff
@@ -0,0 +1,83 @@
+package org.dolphinemu.dolphinemu.dialogs;
+
+import android.app.Dialog;
+import android.os.Bundle;
+
+import org.dolphinemu.dolphinemu.NativeLibrary;
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.activities.EmulationActivity;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+public final class AlertMessage extends DialogFragment
+{
+  private static boolean sAlertResult = false;
+  private static final String ARG_TITLE = "title";
+  private static final String ARG_MESSAGE = "message";
+  private static final String ARG_YES_NO = "yesNo";
+
+  public static AlertMessage newInstance(String title, String message, boolean yesNo)
+  {
+    AlertMessage fragment = new AlertMessage();
+
+    Bundle args = new Bundle();
+    args.putString(ARG_TITLE, title);
+    args.putString(ARG_MESSAGE, message);
+    args.putBoolean(ARG_YES_NO, yesNo);
+    fragment.setArguments(args);
+
+    return fragment;
+  }
+
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(Bundle savedInstanceState)
+  {
+    final EmulationActivity emulationActivity = NativeLibrary.getEmulationActivity();
+    String title = requireArguments().getString(ARG_TITLE);
+    String message = requireArguments().getString(ARG_MESSAGE);
+    boolean yesNo = requireArguments().getBoolean(ARG_YES_NO);
+
+    AlertDialog.Builder builder = new AlertDialog.Builder(emulationActivity,
+            R.style.DolphinDialogBase)
+            .setTitle(title)
+            .setMessage(message);
+
+    // If not yes/no dialog just have one button that dismisses modal,
+    // otherwise have a yes and no button that sets sAlertResult accordingly.
+    if (!yesNo)
+    {
+      setCancelable(false);
+      builder.setPositiveButton(android.R.string.ok, (dialog, which) ->
+      {
+        dialog.dismiss();
+        NativeLibrary.NotifyLock();
+      });
+    }
+    else
+    {
+      sAlertResult = false;
```

Comment:
Is there a reason for this line?

Reply:
I think if a panic alert sets `sAlertResult` to true, a later panic alert will still return sAlertResult = true if a user backs out of the `yesNo` prompt.

Reply:
But doesn't every code path that calls `NotifyAlertMessageLock` set `sAlertResult`? So either there is some way to dismiss the alert without calling `NotifyAlertMessageLock`, or there shouldn't be any need for this line.

Reply:
It turns out directly copying the old code into the DialogFragment didn't handle the case of backing out of a `yesNo` prompt, which never notified the lock. I removed the extra `sAlertResult` and made `yesNo` prompts not cancelable too. This makes sense since users can rapidly tap and accidentally dismiss the prompt.

Reply:
Sounds good to me.